boltdesignsystem/bolt
# Button | FE | Rename props and expand options

## Jira

http://vjira2:8080/browse/BDS-1103
http://vjira2:8080/browse/BDS-1102
http://vjira2:8080/browse/BDS-1101

## Summary

Renamed props within the button component to follow current conventions.

## Details

1. `iconOnly` renamed to `icon_only`.
2. `transform` renamed to `text_transform`.
3. `width` responsive options renamed and expanded: 
    * `full-to-auto@xxsmall`
    * `full-to-auto@xsmall`
    * `full-to-auto@small`
    * I held off on changing this to `display`, that requires more discussions and thoughts.
4. Deprecations are indicated in the twig and schema.
5. Existing PL pages are updated to use the new prop names.

## Deprecations

1. `iconOnly`
    * use `icon_only` instead
2. `transform`
    * use `text_transform` instead
3. `width: full@small`
    * use `full-to-auto@small` instead

## How to test

Run the branch locally and check the button docs. Make sure the schema is right and the demos are displaying the props correctly. Check the twig file for proper deprecations.
